File: RepoTransBench/target_projects/Python/C++/saadmk11_django-newsfeed/tests/original/test_urls.cpp
#include <gtest/gtest.h>
#include <string>
#include <regex>
#include <map>

// Simulated resolver for URL tests
std::map<std::string, std::string> url_map = {
    {"/newsfeed/",                     "newsfeed:latest_issue"},
    {"/newsfeed/issues/",              "newsfeed:issue_list"},
    {"/newsfeed/issues/test-issue/",   "newsfeed:issue_detail"},
    {"/newsfeed/subscribe/",           "newsfeed:newsletter_subscribe"},
    {"/newsfeed/unsubscribe/",         "newsfeed:newsletter_unsubscribe"}
};

std::string resolve(const std::string& url) {
    if (url.find("/newsfeed/subscribe/confirm/") == 0)
        return "newsfeed:newsletter_subscription_confirm";
    auto it = url_map.find(url);
    if (it != url_map.end()) return it->second;
    return "";
}

TEST(TestUrls, resolve_latest_issue) {
    ASSERT_EQ(resolve("/newsfeed/"), "newsfeed:latest_issue");
}
TEST(TestUrls, resolve_issue_list) {
    ASSERT_EQ(resolve("/newsfeed/issues/"), "newsfeed:issue_list");
}
TEST(TestUrls, resolve_issue_detail) {
    ASSERT_EQ(resolve("/newsfeed/issues/test-issue/"), "newsfeed:issue_detail");
}
TEST(TestUrls, resolve_subscribe) {
    ASSERT_EQ(resolve("/newsfeed/subscribe/"), "newsfeed:newsletter_subscribe");
}
TEST(TestUrls, resolve_subscription_confirm) {
    std::string token = "faketoken";
    std::string url = "/newsfeed/subscribe/confirm/" + token + "/";
    ASSERT_EQ(resolve(url), "newsfeed:newsletter_subscription_confirm");
}
TEST(TestUrls, resolve_unsubscribe) {
    ASSERT_EQ(resolve("/newsfeed/unsubscribe/"), "newsfeed:newsletter_unsubscribe");
}

File: RepoTransBench/target_projects/Python/Rust/saadmk11_django-newsfeed/tests/original/test_urls.rs
use crate::urls::resolve;

#[test]
fn test_urls_resolve_latest_issue() {
    let result = resolve("/newsfeed/").unwrap();
    assert_eq!(result, "newsfeed:latest_issue");
}

#[test]
fn test_urls_resolve_issue_list() {
    let result = resolve("/newsfeed/issues/").unwrap();
    assert_eq!(result, "newsfeed:issue_list");
}

#[test]
fn test_urls_resolve_issue_detail() {
    let result = resolve("/newsfeed/issues/any/").unwrap();
    assert_eq!(result, "newsfeed:issue_detail");
}

#[test]
fn test_urls_resolve_subscribe() {
    let result = resolve("/newsfeed/subscribe/").unwrap();
    assert_eq!(result, "newsfeed:newsletter_subscribe");
}

#[test]
fn test_urls_resolve_subscription_confirm() {
    let token = "b8eb9618-eafe-4a98-9463-8925d423dacf";
    let url = format!("/newsfeed/subscribe/confirm/{}/", token);
    let result = resolve(&url).unwrap();
    assert_eq!(result, "newsfeed:newsletter_subscription_confirm");
}

#[test]
fn test_urls_resolve_unsubscribe() {
    let result = resolve("/newsfeed/unsubscribe/").unwrap();
    assert_eq!(result, "newsfeed:newsletter_unsubscribe");
}